src/Functions/ztest.cpp

```diff
@@ -0,0 +1,199 @@
+                "confidence_interval_low",
+                "confidence_interval_high"
+            };
+
+            return std::make_shared<DataTypeTuple>(
+                std::move(types),
+                std::move(names)
+            );
+        }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+
+            for (size_t i = 0; i < 4; ++i)
+            {
+                if (!isNativeInteger(arguments[i]))
+                {
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The {}th Argument of function {} must be a integer or unsigned integer.", i + 1, getName());
+                }
+            }
+
+            if (!isFloat(arguments[4]))
+            {
+                throw Exception{"The fifth argument of function " + getName() + " should be a float, illegal type: " + arguments[4]->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+            }
+
+            if (!isString(arguments[5]))
+            {
+                throw Exception{"The sixth argument of function " + getName() + " should be a string, illegal type: " + arguments[5]->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+            }
+
+            return getReturnType();
+        }
+
+
+        static void nan(MutableColumnPtr & to)
+        {
+            const Float64 nan = std::numeric_limits<Float64>::quiet_NaN();
+            Tuple tuple({nan, nan, nan, nan});
+            to->insert(tuple);
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            auto non_const_arguments = arguments;
+            for (auto & argument : non_const_arguments)
+                argument.column = argument.column->convertToFullColumnIfConst();
+
+            MutableColumnPtr to{getReturnType()->createColumn()};
+            to->reserve(input_rows_count);
+
+            if (!isString(arguments[5].type))
+                throw Exception{"The sixth argument of function " + getName() + " should be String, illegal type: " + arguments[5].type->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+            for (size_t row_num = 0; row_num < input_rows_count; ++row_num)
+            {
+                const UInt64 successes_x = arguments[0].column->getUInt(row_num);
+                const UInt64 successes_y = arguments[1].column->getUInt(row_num);
+                const UInt64 trials_x = arguments[2].column->getUInt(row_num);
+                const UInt64 trials_y = arguments[3].column->getUInt(row_num);
+
+                const Float64 props_x = static_cast<Float64>(successes_x) / trials_x;
+                const Float64 props_y = static_cast<Float64>(successes_y) / trials_y;
+                const Float64 diff = props_x - props_y;
+                const UInt64 trials_total = trials_x + trials_y;
+                const Float64 confidence_level = arguments[4].column->getFloat64(row_num);
```

Comment:
Consider to typeid_cast<ColumnFloat> here, and work with column without indirection